test/Microsoft.Extensions.Hosting.Tests/HostTests.cs

```diff
@@ -414,7 +353,6 @@ public void HostCanBeStarted()
 
                 lifetime = (FakeHostLifetime)host.Services.GetRequiredService<IHostLifetime>();
                 Assert.Equal(1, lifetime.StartCount);
-                Assert.Equal(1, lifetime.StoppingCount);
```

Comment:
Why did this change?

Reply:
Stopping was one of the test callbacks we removed.

Reply:
What do you mean? The test itself changed?

Reply:
I merged RegisterStopCallback and RegisterDelayStartCallback into the new WaitForStartAsync so there's one less event to mock in the tests.

Reply:
Oh, I thought this was IApplicationLifetime